File: bj/7568.py
import sys
n = int(sys.stdin.readline().strip())
points = [1 for _ in range(n)]
records = []

for i in range(n):
    x, y = map(int, sys.stdin.readline().strip().split())
    records.append([x,y])

for j in range(n):
    for k in range(j+1, n):
        if records[j][0] > records[k][0] and records[j][1] > records[k][1]:
            points[k] += 1
        elif records[j][0] < records[k][0] and records[j][1] < records[k][1]:
            points[j] += 1

print(*points)

File: bj/7568.cpp
#include<iostream>
#include<vector>
using namespace std;
int main(){
    int n;
    cin >> n;
    vector<int> points(n, 1);
    vector<vector<int>> records;
    for (int i = 0; i < n; i++){
        int x, y;
        cin >> x >> y;
        records.push_back({x, y});
    }
    for (int j = 0; j < n; j++){
        for (int k = 0; k < n; k++){
            if (records[j][0] > records[k][0] && records[j][1] > records[k][1]){
                points[k] += 1;
            }
            else if (records[j][0] > records[k][0] && records[j][1] > records[k][1]){
                points[j] += 1;
            }
        }
    }
    for (int num:points){
        cout << num << " ";
    }
    return 0;
}